std/cs/internal/Null.hx

```diff
@@ -21,21 +21,33 @@
  */
 package cs.internal;
 
-@:classCode('
-	//This function is here to be used with Reflection, when the haxe.lang.Null type is known
-	public static haxe.lang.Null<T> _ofDynamic(object obj)
-	{
-		if (obj == null)
-		{
+@:classCode('//This function is here to be used with Reflection, when the haxe.lang.Null type is known
+		public static haxe.lang.Null<T> _ofDynamic(object obj) {
+			if (obj == null) {
+				return new haxe.lang.Null<T>(default(T), false);
+			} else if (typeof(T).Equals(typeof(double))) {
+				return new haxe.lang.Null<T>((T) (object) haxe.lang.Runtime.toDouble(obj), true);
+			} else if (typeof(T).Equals(typeof(int))) {
+				return new haxe.lang.Null<T>((T) (object) haxe.lang.Runtime.toInt(obj), true);
+			} else {
+				return new haxe.lang.Null<T>((T) obj, true);
+			}
+		}
+
+		public static implicit operator haxe.lang.Null<T>(T val) {
+			return new haxe.lang.Null<T>(val, true);
+		}
+
+		public static implicit operator T(haxe.lang.Null<T> obj) {
+			return obj.@value;
```

Comment:
When assigning Null<T> to T, Haxe currently generates "t = nt.@value", meaning that it'll return the default value (same as this function) if it's nulled. Can exclude this part from PR.